programs/mango-v4/src/state/oracle.rs

```diff
@@ -317,9 +356,53 @@ pub fn oracle_state_unchecked(
                 oracle_type: OracleType::SwitchboardV1,
             }
         }
+        OracleType::OrcaCLMM => {
+            let usd_state = usdc_state_unchecked(usd_feed_opt)?;
+
+            let whirlpool = Whirlpool::try_deserialize(&mut &data[..]).unwrap();
+
+            let sqrt_price = U64F64::from_bits(whirlpool.sqrt_price);
+
+            let clmm_price = if whirlpool.token_mint_a == usdc_mint_mainnet::ID {
+                // inverted
+                let decimals = QUOTE_DECIMALS - (base_decimals as i8); // tokenMintB - tokenMintA
+                let decimal_adj = power_of_ten(decimals);
+                let inv_price = I80F48::from_num(sqrt_price * sqrt_price) * decimal_adj;
+                Ok(I80F48::ONE.checked_div(inv_price).unwrap())
+            } else if whirlpool.token_mint_b == usdc_mint_mainnet::ID {
+                let decimals = (base_decimals as i8) - QUOTE_DECIMALS; // tokenMintA - tokenMintB
+                let decimal_adj = power_of_ten(decimals);
+                Ok(I80F48::from_num(sqrt_price * sqrt_price) * decimal_adj)
+            } else {
+                Err(MangoError::InvalidCLMMOracle)
+            }?;
+
+            let price = clmm_price * usd_state.price;
+            OracleState {
+                price,
+                last_update_slot: u64::MAX, // REVIEW: should we inherit pyth feed confidence metrics?
```

Comment:
I think this should be `min(pyth.last_update_slot, orca.last_update_slot)` - and if we don't have one for orca, then just pyth

Reply:
sadly, orca does not store last update slot